src/distributors/zulu-installer.ts

```diff
@@ -0,0 +1,73 @@
+import * as core from '@actions/core';
+import * as tc from '@actions/tool-cache';
+
+import path from 'path';
+import fs from 'fs';
+import semver from 'semver';
+
+import { JavaInstallerResults, IJavaRelease, JavaBase, JavaInstallerOptions } from './base-installer';
+import { IZulu, IZuluDetailed } from './zulu-models';
+import { IS_WINDOWS, IS_MACOS, PLATFORM } from '../util';
+
+export class ZuluDistributor extends JavaBase {
+    private extension = IS_WINDOWS ? 'zip' : 'tar.gz';
+    private platform: string;
+    constructor(initOptions: JavaInstallerOptions) {
+        super("Zulu", initOptions);
+        this.platform = IS_MACOS ? 'macos' : PLATFORM;
+        this.arch = this.arch === 'x64' ? 'x86' : this.arch;
+    }
+
+    protected async downloadTool(javaRelease: IJavaRelease): Promise<JavaInstallerResults> {
+        let toolPath: string;
+        let extractedJavaPath: string;
+
+        const javaArchivePath = await tc.downloadTool(javaRelease.link);
+        
+        core.info(`Ectracting ${this.distributor} java version ${javaRelease.resolvedVersion}`);
+        if(IS_WINDOWS) {
+            extractedJavaPath = await tc.extractZip(javaArchivePath);
+        } else {
+            extractedJavaPath = await tc.extractTar(javaArchivePath);
+        }
+
+        const archiveName = fs.readdirSync(extractedJavaPath)[0];
+        const archivePath = path.join(extractedJavaPath, archiveName);
+        toolPath = await tc.cacheDir(archivePath, `Java_${this.distributor}_${this.javaPackage}`, javaRelease.resolvedVersion, this.arch);
+
+        return { javaPath: toolPath, javaVersion: javaRelease.resolvedVersion };
+    }
+
+    protected async resolveVersion(range: semver.Range): Promise<IJavaRelease> {
```

Comment:
may be rename `range` to `version` here and other places?

Reply:
Done.